code/crates/engine/src/wal/entry.rs

```diff
@@ -80,11 +80,8 @@ where
             }
 
             WalEntry::Timeout(timeout) => {
-                // Write tag
-                buf.write_u8(Self::TAG_TIMEOUT)?;
-
-                // Write timeout
-                encode_timeout(timeout, &mut buf)?;
+                // Write tag and timeout if applicable
+                encode_timeout(Self::TAG_TIMEOUT, timeout, &mut buf)?;
```

Comment:
Nice catch! Thanks so much for figuring this out 🙌

Reply:
Wondering if we should have consensus decide which events should be logged and have wal write everything it's told to (by consensus).

Reply:
80afd66d6d78321b39a9e65442a75c1c2a880421 + ea04fba76a3df74089432a607d494a51a268b0b3